pkg/plugins/removerplugin.go

```diff
@@ -0,0 +1,89 @@
+package plugins
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/kustomize/api/resmap"
+	"sigs.k8s.io/kustomize/api/resource"
+	"sigs.k8s.io/kustomize/kyaml/kio"
+	kyaml "sigs.k8s.io/kustomize/kyaml/yaml"
+	"sigs.k8s.io/yaml"
+)
+
+type RemoverPlugin struct {
+	Gvk  schema.GroupVersionKind
+	Path []string
+}
+
+var _ resmap.Transformer = &RemoverPlugin{}
+
+type RemoverFilter struct {
+	Gvk  schema.GroupVersionKind
+	Path []string
+}
+
+var _ kio.Filter = RemoverFilter{}
+
+func (f RemoverFilter) Filter(nodes []*kyaml.RNode) ([]*kyaml.RNode, error) {
+	return kio.FilterAll(kyaml.FilterFunc(f.run)).Filter(nodes)
+}
+
+func (f RemoverFilter) run(node *kyaml.RNode) (*kyaml.RNode, error) {
+	if len(f.Path) == 0 {
+		return node, fmt.Errorf("empty path")
+	}
+
+	typeMeta := kyaml.TypeMeta{
+		APIVersion: f.Gvk.GroupVersion().String(),
+		Kind:       f.Gvk.Kind,
+	}
+
+	meta, err := node.GetMeta()
+	if err != nil {
+		return node, err
+	}
+
+	if meta.TypeMeta != typeMeta {
+		return node, nil
+	}
+
+	pathLen := len(f.Path)
+	path := f.Path[:pathLen-1]
+	name := f.Path[pathLen-1]
+
+	matcher := &kyaml.PathMatcher{Path: path}
+	result, err := node.Pipe(matcher)
+	if err != nil {
+		return node, err
+	}
+
+	return node, result.VisitElements(
+		func(node *kyaml.RNode) error {
+			return node.PipeE(kyaml.FieldClearer{Name: name})
+		})
+}
+
+func (p *RemoverPlugin) Config(_ *resmap.PluginHelpers, c []byte) error {
+	p.Gvk = schema.GroupVersionKind{}
+	p.Path = nil
+	return yaml.Unmarshal(c, p)
+}
+
+func (p *RemoverPlugin) Transform(m resmap.ResMap) error {
+	return m.ApplyFilter(RemoverFilter{
+		Gvk:  p.Gvk,
+		Path: p.Path,
+	})
```

Comment:
```suggestion

	filter := RemoverFilter{
		Gvk:  p.Gvk,
		Path: p.Path,
	}
	return m.ApplyFilter(filter)
```
just to be consistent as in TransformResource